x/txfees/keeper/fees.go

```diff
@@ -167,5 +166,25 @@ func (k Keeper) swapFeeToBaseDenom(
 		return nil, sdk.Coins{takerFeeCoin}, nil
 	}
 
+	// If the swap is successful, we add the taker fee attribute to the emitted event
+	k.appendTakerFeeAttribute(ctx)
+
 	return sdk.Coins{{Denom: baseDenom, Amount: tokenOutAmount}}, nil, nil
 }
+
+// appendTakerFeeAttribute modifies the last token swap event to include the taker fee attribute
+func (k Keeper) appendTakerFeeAttribute(ctx sdk.Context) sdk.Context {
+	emittedEvents := ctx.EventManager().Events()
+	if len(emittedEvents) > 0 && emittedEvents[len(emittedEvents)-1].Type == gammtypes.TypeEvtTokenSwapped {
+		ev := emittedEvents[len(emittedEvents)-1].AppendAttributes(sdk.NewAttribute(AttributeKeyTakerFee, "true"))
+
+		emittedEvents = emittedEvents[:len(emittedEvents)-1]
+		emittedEvents = append(emittedEvents, ev)
+
+		ctx = ctx.WithEventManager(sdk.NewEventManager())
```

Comment:
why do we need this line?

Reply:
it clears the events before adding the again after modification